src/tables.jl

```diff
@@ -0,0 +1,69 @@
+"""
+Constructs `Table` out of `FeatureLayer`, where every row is a `Feature` consisting of Geometry and attributes.
+```
+ArchGDAL.Table(T::Union{IFeatureLayer, FeatureLayer})
+```
+"""
+struct Table{T<:Union{IFeatureLayer, FeatureLayer}}
+    layer::T
+end
+
+getlayer(t::Table) = Base.getfield(t, :layer)
+
+function Tables.schema(layer::AbstractFeatureLayer)
+    featuredefn = layerdefn(layer)
+    fielddefns = (getfielddefn(featuredefn, i) for i in 0:nfield(layer)-1)
+    names_fields = (getname(fielddefn) for fielddefn in fielddefns)
+    types_fields = (_FIELDTYPE[gettype(fielddefn)] for fielddefn in fielddefns)
+    
+    ngeom = ArchGDAL.ngeom(featuredefn)
+    geomdefns = (ArchGDAL.getgeomdefn(featuredefn, i) for i in 0:ngeom-1)
+    geom_names = (ArchGDAL.getname(geomdefn) for geomdefn in geomdefns)
+    geom_types = (IGeometry for i in 1:ngeom)
```

Comment:
Consider defining a method like

```julia
function schema_names(featuredefn)
    fielddefns = (getfielddefn(featuredefn, i) for i in 0:nfield(layer)-1)
    field_names = Tuple(Symbol(getname(fielddefn)) for fielddefn in fielddefns)
    geom_names = Tuple(Symbol(getname(getgeomdefn(featuredefn, i-1))) for i in 1:ngeom(layer))
    return (field_names, geom_names)
end
```

and using it across both here and in the `get_rows()` function.

Reply:
Alternatively, use `Tables.schema(layer)` in `function get_row(layer::IFeatureLayer)` later.